include/numerics/petsc_matrix.h

```diff
@@ -239,6 +239,19 @@ class PetscMatrix final : public SparseMatrix<T>
    */
   virtual void add (const T a, const SparseMatrix<T> & X) override;
 
+  /**
+   * Compute Y = A*X for matrix \p X.
+   */
+  void mat_mult (SparseMatrix<T> & X, SparseMatrix<T> & Y);
+
+  /**
+   * Add \p scalar* \p spm to the rows and cols of this matrix
```

Comment:
Could we expand this explanation a bit?  Something like ```A_rows[i]cols[j] += scalar * spm_ij```?  "to the rows and cols" seems ambiguous until you go look at the implementation.

Reply:
Good point! This has been taken care of.